### PR TITLE
Indentation of deadline

### PR DESCRIPTION
**Pull request by paretje**
In my org file, I had some deadlines, which weren't recognized. I soon found the problem to be indentation. Recently, I've checked the org manual, and I think it's allowed everywhere, except footnotes, which aren't implemented yet.

### CODE REVIEW

File: autoload/dotoo/parser/lexer.vim [hunk 1]
```diff
@@ -29,15 +29,15 @@ let s:todo_keywords_todo = g:dotoo#parser#todo_keywords[:index(g:dotoo#parser#to
 let s:todo_keywords_done = g:dotoo#parser#todo_keywords[index(g:dotoo#parser#todo_keywords,'|'):]
 let s:todo_keywords_regex = join(s:todo_keywords_todo+s:todo_keywords_done, '|')
 call s:define('blank', '\v^$')
-call s:define('directive', '\v^#\+(\w+): (.*)$')
+call s:define('directive', '\v^\s*#\+(\w+): (.*)$') " ik denk van wel
```
Comment (dhruvasagar) [inline autoload/dotoo/parser/lexer.vim]: Can you translate the comment in english please ?

**Comment by dhruvasagar** [comment]
Since you've updated the parser token regexes, this would parse correctly, but it doesn't take those indents into account while building the object tree, so while serializing the content the indents will be lost.

**Comment by paretje** [comment]
Indeed, the most important thing is that these indents, which are correct, are supported, especially with auto-indentation. Keeping indentation after serialization is more "difficult". Unless we would explicitly keep this information, it's possible indentation changes.

I'm not sure what you would prefer: keeping the current situation without indentation, or with?

**Comment by dhruvasagar** [comment]
Personally I don't care about indentation, but if we are going to support it, then we need the parser and object builders to recognize it and incorporate while serializing. It's pretty easy to do really, just didn't think it would be needed but I suppose it doesn't hurt.

**Comment by paretje** [comment]
OK, now the serialization does indent. When I said it's "difficult", I meant there isn't really a good way to keep the indentation used by the user. It's easy if you just use a general convention.

**Comment by dhruvasagar** [comment]
Hi,

I have been a bit busy travelling and such. On the outset the diffs look good except it looks like we're kinda forcing the indents to be a certain value as per the headline level in the heirarchy.

I'll have to play around with this and see how it looks, however the 1 thing missing in this is that I feel indentation support should be configurable, i.e. the user should be able to enable / disable it.

I'll look more into this, especially how we can make it user configurable.

**Comment by paretje** [comment]
Well, it follows to rules of the autoindentation. Indentation (of text) is kept intact when moving headlines, and thus, everything ends up using the indentation of first level headlines.

Fixing indentation wouldn't be a good idea, I think, as you might even consider fixing according textwidth then. I think it should be either kept as is, and let the user fix layout of the actual content if he so wishes, or there should only be support to parse indentation, as was my original intention.

**Comment by dhruvasagar** [comment]
Hey, I am going to get more active now, had been travelling a lot. Planning a milestone release by end of december.

**Comment by dareenzo** [comment]
I have been trying to get this up and running and run into the issue this PR tackles. My `DEADLINE`s and `SCHEDULE`s were indented and thus the entries didn't show up on the agenda view!

**Comment by dhruvasagar** [comment]
Yes and I think there are other subtle issues with this, why it hasn't been merged. Although it technically should be fairly easy to implement this.

**Comment by dhruvasagar** [comment]
Indentation requires a bit more effort since this & some partial attempts i've made in the past broke a few things, I am going to try to do this from scratch.

**Pull request closed without merging**